frame/election-provider-support/src/lib.rs

```diff
@@ -369,79 +367,55 @@ pub trait ElectionProviderBase {
 	/// The error type that is returned by the provider.
 	type Error: Debug;
 
+	/// The upper bound on election winners that cab be returned.
+	///
+	/// # WARNING
+	///
+	/// when communicating with the data provider, one must ensure that
+	/// `DataProvider::desired_targets` returns a value less than this bound. An implementation can
+	/// chose to either return an error and/or sort and truncate the output to meet this bound.
+	type MaxWinners: Get<u32>;
+
 	/// The data provider of the election.
 	type DataProvider: ElectionDataProvider<
 		AccountId = Self::AccountId,
 		BlockNumber = Self::BlockNumber,
 	>;
-
-	/// Indicate if this election provider is currently ongoing an asynchronous election or not.
-	fn ongoing() -> bool;
 }
 
-/// Same as `BoundedSupports` but parameterized by a `BoundedElectionProvider`.
-pub type BoundedSupportsOf<E> = BoundedSupports<
-	<E as ElectionProviderBase>::AccountId,
-	<E as BoundedElectionProvider>::MaxWinners,
->;
-
 /// Elect a new set of winners, bounded by `MaxWinners`.
 ///
-/// Returns a result in bounded, target major format, namely as
-/// *BoundedVec<(AccountId, Vec<Support>), MaxWinners>*.   
-pub trait BoundedElectionProvider: ElectionProviderBase {
-	/// The upper bound on election winners.
-	type MaxWinners: Get<u32>;
+/// It must always use `Config::DataProvider` to fetch the data it needs.
+///
+/// This election provider that could function asynchronously. This implies that this election might
+/// needs data ahead of time (ergo, receives no arguments to `elect`), and might be `ongoing` at
+/// times.
+pub trait ElectionProvider: ElectionProviderBase {
```

Comment:
As I tried to demonstrate in docs, `ElectionProvider` is async by default, `InstantElectionProvider` is marker that specifies it is sync. They both use the same `DataProvider`.